### DISCUSSION TITLE
Change the compiler on `filetypes.cs` doesn't work

### DISCUSSION THREAD

**Discussion by ghost** [geany/geany#3753] [General]
It appeared to be hardcoded `mcs` as the compiler.

**Comment by elextr** [comment]
How did you "change the compiler" I have no clue what you did wrong without further information.

**Reply by ghost** [reply]
Edit `filetypes.cs` and change the compiler from `mcs` to `csc`? I keep the `filetypes.cs` on Geany installation directory intact. Instead I copy it into `%appdata%\geany\filedefs` and modify the version here. It doesn't work. I even tried to comment out `C#=*.cs;` in `%appdata%\geany\filetype_extensions.conf`. Nothing worked. It will always force `mcs` as the compiler.

**Comment by elextr** [comment]
Why not use "Menu->Build->Set build commands" in the GUI.  Remove all the changes and extra files and try that.

**Reply by ghost** [reply]
Tried. Not work. `mcs` is hardcoded as the compiler.

**Reply by ghost** [reply]
As far as I know, the setting via "Menu->Build->Set build commands" is per file or per session. I want to change the compiler to `csc` for all and, better yet, out of the box by just copying a `filetypes.cs` file into `%appdata%\geany\filedefs`.

**Comment by ghost** [comment]
What worked:

Edit `filetypes.cs` and remove this block:

```
[build_settings]
# %f will be replaced by the complete filename
# %e will be replaced by the filename without extension
# be careful following settings are untested
compiler=mcs /t:winexe "%f" /r:System,System.Drawing
run_cmd=mono "%e.exe"
```

Replace it with something like this:

```
[build-menu]
FT_00_LB=_Compile
FT_00_CM=csc /t:winexe "%f" /r:System,System.Drawing
FT_00_WD=
FT_01_LB=_Build
FT_01_CM=csc /t:winexe "%f" /r:System,System.Drawing
FT_01_WD=
EX_00_LB=_Run
EX_00_CM="./%e"
EX_00_WD=
```

**Comment by elextr** [comment]
> As far as I know, the setting via "Menu->Build->Set build commands" is per file or per session.

No, its per user, and as such should override settings in the filetype file.  Did you have a project open?  That also has build commands.

See [the wiki](https://wiki.geany.org/howtos/configurebuildmenu) for details.
